crates/sui-core/src/authority_active.rs

```diff
@@ -218,6 +237,41 @@ where
         })
     }
 
+    pub async fn sync_to_latest_checkpoint(&self) -> SuiResult {
+        self.sync_to_latest_checkpoint_with_config(Default::default())
+            .await
+    }
+
+    pub async fn sync_to_latest_checkpoint_with_config(
+        &self,
+        checkpoint_process_control: CheckpointProcessControl,
+    ) -> SuiResult {
+        let checkpoint_store =
+            self.state
+                .checkpoints
+                .clone()
+                .ok_or(SuiError::UnsupportedFeatureError {
+                    error: "Checkpoint not supported".to_owned(),
+                })?;
+
+        let (checkpoint_summary, _) = get_latest_proposal_and_checkpoint_from_all(
```

Comment:
In the fullnode path, it's unnecessary to get proposals. We may want to separate proposal fetch and checkpoint fetch.

Reply:
Ok, I was wondering about this when I wrote this. I will add a TODO and handle it in the next PR.